tests/test_cases/issue_134/test_reals.py

```diff
@@ -26,3 +26,21 @@ def assign_double(dut):
     log.info("Read back value %g" % got)
     if got != val:
         raise TestFailure("Values didn't match!")
+
+@cocotb.test(expect_error=cocotb.SIM_NAME in ["Icarus Verilog"])
+def assign_double(dut):
+    """
+    Assign a random floating point value, read it back from the DUT and check
+    it matches what we assigned
+    """
+    val = random.randint(-2**31, 2**31 - 1)
+    log = logging.getLogger("cocotb.test")
+    yield Timer(1)
+    log.info("Setting the value %i" % val)
+    dut.stream_in_real <= val
+    yield Timer(1)
+    yield Timer(1) # Workaround for VHPI scheduling - needs investigation
+    got = int(dut.stream_out_real)
```

Comment:
Better not to cast it to int here - you want to check that writing `1` did not write `1.2` or something

Reply:
Sure. I'm assuming we model these in double precision. Since my integers are in the range `[-2**31, 2**31-1]`, this will only be true if the mantissa is > 32 bits (which is true for double, but not single precision).